src/graphnet/models/model.py

```diff
@@ -164,6 +194,14 @@ def predict_as_dataframe(
             additional_attributes = []
         assert isinstance(additional_attributes, list)
 
+        if prediction_columns is None:
+            try:
+                prediction_columns = self.prediction_columns
+            except AttributeError:
+                assert (
+                    1 == 2
+                ), "Could not infer prediction_columns from model. Please specify prediction_columns."
+        assert isinstance(prediction_columns, list)
```

Comment:
I don't think this is a great solution. 😊  If we're assuming the existence of the property `self.prediction_columns` in `Model.predict_as_dataframe` then we should ensure that it is set for all instances of `Model`.

You could do something like:
```python
class Model:
    def __init__(self, <...>, prediction_columns: Optional[List[str]] = None, ...):
        <...>
        self._prediction_columns = prediction_columns

    @property
    def prediction_columns(self) -> Optional[List[str]]:
        return self._prediction_columns

    @property
    def target_columns(self) -> Optional[List[str]]:
        raise NotImplementedError
```
This means that all instances of `Model` have the property `self.prediction_columns`, and that instances of `StandardModel` override the two properties `self.prediction_columns` and `self.target_columns` in a way that is meaningful for their implementation.